packages/sdk/src/Actions/EventAction.ts

```diff
@@ -220,15 +227,15 @@ export interface ActionStep {
  * Parameters for validating an action step.
  *
  * @typedef {Object} ValidateActionStepParams
- * @property {Record<Hex, AbiEvent>} [knownEvents] - Optional record of known events, keyed by their hex signature.
- * @property {AbiEvent} [event] - Optional ABI event definition.
+ * @property {Record<Hex, AbiEvent | AbiFunction>} [knownSignatures] - Optional record of known events, keyed by 32 byte selectors.
+ * @property {AbiEvent | AbiFunction} [abiItem] - Optional ABI item definition.
  * @property {EventLogs} [logs] - Event logs to validate against. Required if 'hash' is not provided.
  * @property {Hex} [hash] - Transaction hash to validate against. Required if 'logs' is not provided.
  * @property {number} [chainId] - Chain ID for the transaction. Required if 'hash' is provided.
  */
 export type ValidateActionStepParams = {
-  knownEvents?: Record<Hex, AbiEvent>;
-  event?: AbiEvent;
+  knownSignatures?: Record<Hex, AbiEvent | AbiFunction>;
+  abiItem?: AbiEvent | AbiFunction;
```

Comment:
Awesome ty for adding function support

Reply:
ty for doing literally all the hard work and letting me ctrl+c your work